src/com/puzzleitc/jenkins/command/OpenshiftProcessCommand.groovy

```diff
@@ -0,0 +1,57 @@
+package com.puzzleitc.jenkins.command
+
+import com.puzzleitc.jenkins.command.context.PipelineContext
+
+import static com.puzzleitc.jenkins.command.Constants.DEFAULT_OC_TOOL_NAME
+
+class OpenshiftProcessCommand {
+
+    private static final VALID_OUTPUT_FORMAT = ['json', 'yaml']
+
+    private final PipelineContext ctx
+
+    OpenshiftProcessCommand(PipelineContext ctx) {
+        this.ctx = ctx
+    }
+
+    Object execute() {
+        ctx.info('-- openshiftProcess --')
+        def result = null
+        def templateFilePath = ctx.stepParams.getRequired('templateFilePath')
+        def params = ctx.stepParams.getOptional('params')
+        def paramFile = ctx.stepParams.getOptional('paramFile') as String
+        def ignoreUnknownParameters = ctx.stepParams.getOptional('ignoreUnknownParameters', false) as Boolean
+        def labels = ctx.stepParams.getOptional('labels')
+        def output = ctx.stepParams.getOptional('output', 'json') as String
+
+        validateOutputFormat(output)
+        ctx.echo("template file: ${templateFilePath}")
+        ctx.withEnv(["PATH+OC=${ctx.executable('oc', DEFAULT_OC_TOOL_NAME)}"]) {
+
+            def processScript = 'oc process' + ' ' + '-f ' + templateFilePath
+            if (params) {
+                processScript = processScript + ' ' + params.join(' ')
+            }
+            if (labels) {
+                processScript = processScript + ' -l ' + labels.join(' -l ')
+            }
+            if (paramFile) {
+                processScript = processScript + ' ' + '--param-file=' + paramFile
+            }
+            processScript = processScript + ' ' +
+                '--local=true' + ' ' +
+                '--ignore-unknown-parameters=' + ignoreUnknownParameters + ' ' +
+                '--output=' + output
+            result = ctx.sh(script: processScript, returnStdout: true)
```

Comment:
Minor: Ich fände es leserlicher, den Returnwert direkt zurückzugeben, statt ihn zuerst einer Variable zuzuweisen.

Reply:
Das mit dem Return wollte ohne diese Zuweisung beim Testen nichts zurück geben.
In Absprache mit dbruegger geschlossen.